packages/@lwc/integration-karma/test/accessibility/LightningElement.focus/index.spec.js

```diff
@@ -80,8 +80,29 @@ describe('LightningElement.focus', () => {
             expect(elm.shadowRoot.activeElement).toBeNull();
         });
 
-        // TODO [#2566]: Firefox behaves differently from Chrome/Safari in this test
-        if (!(process.env.NATIVE_SHADOW && delegatesFocus)) {
+        // TODO [#3724]: There's a bug in Safari that causes non-focusable custom elements
+        // to be focusable when formAssociated=true.
+        // Note in Chrome and Firefox the behavior is the same, document.activeElement does not point to the
+        // custom element but to the <body>
+        const customElementFocusableWhenFormAssociated = () => {
+            if (!customElements.get('safari-focus-bug')) {
+                customElements.define(
+                    'safari-focus-bug',
+                    class SafariFocusBug extends HTMLElement {
+                        static formAssociated = true;
+                    }
+                );
+            }
+            const elm = document.createElement('safari-focus-bug');
+            document.body.appendChild(elm);
+            elm.focus();
+            return document.activeElement === elm;
+        };
```

Comment:
@nolanlawson can I get your thoughts on whether this is a good way to test for the Safari bug?

I couldn't find a reliable way to detect the browser versions so I'm testing to see if the bug behavior exists instead.

Reply:
This works perfectly. Well done!

I would just be sure to remove the `elm` when you're done.

Reply:
Shouldnt an assertion be done before elm.focus() to prove it is different

Reply:
@AllanOricil It's impossible for it to be active until `elm.focus()` is called

In any case, I copy-pasted this code into all 3 browsers' DevTools, and it was only `true` for Safari :)

Reply:
@AllanOricil hrrm that's a good point, it would be good to ensure where the focus is before the validation runs. I added a line to reset focus before the validation begins, thanks!